### PR TITLE
Filters for WW powheg mass bins

### PR DESCRIPTION
**Pull request by dnash86**
I actually have a few questions regarding these cards with filters, since I have measured oddly low efficiencies for the filters with these and the tarballs produced from the WW cards in https://github.com/cms-sw/genproductions/pull/473.

### CODE REVIEW

File: python/ThirteenTeV/Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_WW_Mee_1200_2500.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                                    'processParameters'
+                                    )
+        )
+                         )
+
+
+llgenfilter = cms.EDFilter("MCParticlePairFilter",
+                           Status = cms.untracked.vint32(23, 23),
+                           MinPt = cms.untracked.vdouble(0., 0.),
+                           #ParticleCharge = cms.untracked.int32(0),
```
Comment (dnash86) [inline python/ThirteenTeV/Hadronizer_TuneCUETP8M1_13TeV_powhegEmissionVeto_WW_Mee_1200_2500.py]: I have tried these both with and without this line.  Both times I got extremely low efficiencies.  Is this filter configured correctly or have I done something wrong?

**Comment by dnash86** [comment]
Filter efficiency for Mee_200_600: (119) / (5000).

This strikes me as quite low.  Is there perhaps a problem here?

**Comment by covarell** [comment]
I would also leave out the Status line from the filter

**Comment by bendavid** [comment]
@covarell, on this I'm not sure.  There can be many copies of the W's, so if the status line is left out, then there will effectively be a giant OR on the kinematics at all stages of the parton shower (and the samples will not be mutually exclusive)

It should be verified that there are exactly two status 22 W's in every event, then it should be correct to explicitly select on status 22.  (but indeed, status 23 is not correct, since this is only for outgoing matrix element particles, not decayed ones!)

**Comment by covarell** [comment]
But the filter is on electrons, not W...
So for me it is fine to have no Status line, or even better Status = (1, 1)

**Comment by bendavid** [comment]
Sorry, what exactly is the filter supposed to select?

**Comment by dnash86** [comment]
Sorry, I thought I had mentioned what the filter should be on.  It is actually on M_ll (since this is a major background for Z').  Is status (1,1) okay for pythia8?  I thought that was a pythia6 status?

**Comment by bendavid** [comment]
So in principle status 1 can also include non-prompt leptons.  In practice it's probably mostly ok for harder cuts.
The proper solution cannot be done solely with status codes, and requires the gen status flags, which are not available in this filter.  So that would require additional code changes.
If you want to be 100% sure there is no overlap between the samples, then this would be the way to go.  In practice the difference is probably small.

For tau's, if you want to do it only based on status codes, you would need to select on status 2 tau's (with the same caveats as above, that there can be non-prompt taus from hadron decays, etc in principle)

**Comment by dnash86** [comment]
Overlap between samples is okay, as the main reason for these bins is to ensure high statistics are available at higher M_ll values.  The issue was that I was surprised by the low efficiency of this filter.

I can try with status(1,1) for ee and mumu and status (2,2) for tautau states to see what efficiency that yields.  

Would the possible negative weights issue brought up in https://github.com/cms-sw/genproductions/pull/473 have any effect here?  The efficiency taking weights into account was the same as without taking weights into account.

**Comment by bendavid** [comment]
Negative weights should be negligible for this process.

And yes, the problem is that status 23 leptons may not always appear depending on the event history,
so status 1/2 would be the way to go.
(Those two status codes are universal)

**Comment by dnash86** [comment]
I tried status(1,1) for ee and mumu states and still get some fairly low efficiencies:

(394) / (50000) for the 600 < M_ll < 1200 bin, for example.

I see this message in the output, as well:

"WARNING: MCParticlePairFilter : size of some vectors not matching with 2!!"

Is that indicative of an issue with the pair filter for this process and cut choice?  Or should very low efficiencies be expected?

**Comment by bendavid** [comment]
I would suggest to compare this to what you get selecting by hand on the
leptons in the LHEEventProduct to be sure whether it is reasonable or not.

**Comment by covarell** [comment]
In fact, I think m_ll in WW peaks at quite low values, so I am not totally surprised of < 1% for M_ll > 600. Why are you surprised?

**Comment by dnash86** [comment]
I was surprised because the requested bins go quite high (~6000 GeV).  Thus, event counts at GEN level would have to be *quite* high to achieve reasonable statistics (~100k per bin) in the higher bins.  

But it seems that you might be right, and everything may be fine with the filter (although I guess then the GEN level requests will require very high event counts).  I saved the root file from one point (I wasn't before) and checked the LHEEventProduct.isPresent bool:

There were 238 entries in "isPresent", and the filter yielded this efficiency - (238) / (50000).  So it does seem to match.

**Comment by bendavid** [comment]
What is the filter efficiency for the highest mass bin?

**Comment by dnash86** [comment]
At the event counts I was running, it was zero.  Hence my concern something was wrong.

**Comment by dnash86** [comment]
0/50000

**Comment by bendavid** [comment]
OK, well without the ability to apply cuts at the powheg level I think the request is basically impossible then.

**Comment by dnash86** [comment]
Okay, looking in to what to be done about this.

Will upload filters for the ttbar processes momentarily.

**Comment by bendavid** [comment]
In general it is much easier to do this in madgraph (either at LO or NLO) since the cuts can be defined at the LHE level in that case.

**Comment by dnash86** [comment]
Yes, madgraph may have to be the fall back.  Requesters originally wanted powheg.

**Pull request closed without merging**